### PR TITLE
Sockets

### LINKED ISSUE #12: Implement Socket Client
- [x] Import Socket Client and get baseline functionality / server comms up

### PR DESCRIPTION
**Pull request by GreyMatteOr**
## Description
Clearly and concisely describe the feature.

Did some minor cleanup of no longer necessary functions as well as consolidated some state-vars and relegated them to the backend.
  - Anything related to API calls
  - some logic booleans for navigation

## Context

  - Implement sockets.io

## Where To Start

  - Pull down the latest bangwords server and run it with `node server.js`.
  - launch 1 instance of the webpage with `npm start`
  - launch a 2nd complete instance in a separate terminal window

try to play the game with the browser windows side-by-side

## Affected areas of application

- The whole thing

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] Run in browser using the above list method

## Relevant Tickets

closes #12

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -26,43 +28,55 @@ export class App extends Component{
...
-      // dont need numPlayers, instead need boolean "ready"
-      // the below line will get cleaned so we can just setState({joinGame})
-      this.setState({ isGenerator: joinGame.isGen, numPlayers: joinGame.numPlayers, id: joinGame.id })
-    } catch(error) {
-      console.log('error', error)
-    }
-    // console.log('role', role)
+  joinGame = (role) => {
+    client.emit('joinGame', role);
   }
```
Comment (holladayian) [inline src/App.js]: This is so cool!

File: src/App.js [hunk 2]
```diff
@@ -41,72 +35,51 @@ export class App extends Component{
...
     }
-    if (this.state.numPlayers > 1 && !this.state.word && History[History.length -1] !== '/word-selector') {
-      window.setTimeout(() => History.push('/word-selector'), 1);
-    } else if (this.state.numPlayers > 1 && this.state.word && History[History.length -1] !== '/gamepage') {
-
-      window.setTimeout(() => History.push('/gamepage'), 1);
-    } else if (History[History.length -1] !== '/') {
-      window.setTimeout(() => History.push('/'), 1);
-    }
+    this.setHistory()
```
Comment (holladayian) [inline src/App.js:81]: `this.setHistory()`? Does the seriously replace all of the conditional foolery from above?

File: src/Gamepage/Gamepage.js [hunk 3]
```diff
@@ -26,8 +26,9 @@ export class Gamepage extends Component {
   }
 
   mapAttempts = () => {
+    console.log(this.state.attempts)
     let theKey = 0;
-    return this.props.attempts.map(attempt => {
+    return this.state.attempts.map(attempt => {
```
Comment (holladayian) [inline src/Gamepage/Gamepage.js:31]: ?? Are the attempts now stored in Gamepage?

File: src/App.js [hunk 4]
```diff
@@ -3,96 +3,82 @@ import { Switch, Route } from 'react-router-dom';
...
+      this.setState(state);
+    });
+
+    client.on( 'newWordToGuess', (state) => {
+      this.setState(state);
+    });
+
+    client.on( 'result', (state) => {
+      this.setState(state);
+    });
```
Comment (Chriscastanuela) [inline src/App.js:39]: 🔥

**Review by Chriscastanuela — APPROVED** [review]

File: src/Homepage/Homepage.js [hunk 5]
```diff
@@ -6,19 +6,21 @@ export const Homepage = (props) => {
...
         <button
-        className="generator"
-        onClick={() => {
-          props.designateRole('true')
-        }
-        }
-        >Generator</button>
+          className="generator"
+          onClick={() => {
+            props.designateRole(true)
```
Comment (holladayian) [inline src/Homepage/Homepage.js:11]: So these didn't have to get a string after all? BTW this reminds me. We need to go back through and add prop types

**Review by holladayian — APPROVED** [review]

**Pull request merged**